packages/effects/src/dapps/flux/getFluxCdps.ts

```diff
@@ -0,0 +1,282 @@
+          };
+        }
+
+        // Get NFT data for all CDPs at specific state version
+        const nftDataResults = yield* Effect.all(
+          allNftIds.map((nftId) =>
+            entityNonFungibleDataService({
+              resource_address: FluxConstants.receiptResourceAddress,
+              non_fungible_ids: [nftId],
+              at_ledger_state: input.at_ledger_state,
+            })
+          )
+        );
+
+        // Get interest rate keys from both KVS
+        const [xrdKeys, lsulpKeys] = yield* Effect.all([
+          keyValueStoreKeysService({
+            key_value_store_address: FluxConstants.xrdKvsAddress,
+            at_ledger_state: input.at_ledger_state,
+          }),
+          keyValueStoreKeysService({
+            key_value_store_address: FluxConstants.lsulpKvsAddress,
+            at_ledger_state: input.at_ledger_state,
+          }),
+        ]);
+
+        // Get interest rate data from both KVS
+        const [xrdInterestData, lsulpInterestData] = yield* Effect.all([
+          keyValueStoreDataService({
+            key_value_store_address: FluxConstants.xrdKvsAddress,
+            keys: xrdKeys.items.map((k: any) => ({
+              key_json: k.key.programmatic_json,
+            })),
+            at_ledger_state: input.at_ledger_state,
+          }),
+          keyValueStoreDataService({
+            key_value_store_address: FluxConstants.lsulpKvsAddress,
+            keys: lsulpKeys.items.map((k: any) => ({
+              key_json: k.key.programmatic_json,
+            })),
+            at_ledger_state: input.at_ledger_state,
+          }),
+        ]);
+
+        // Process CDP data
+        const cdps: FluxCdpPosition[] = [];
+
+        for (let i = 0; i < allNftIds.length; i++) {
+          const nftId = allNftIds[i];
+          const nftData = nftDataResults[i];
+
+          if (!nftData.non_fungible_ids[0]?.data?.programmatic_json) continue;
+
+          const parsed = CdpNftData.safeParse(
+            nftData.non_fungible_ids[0].data.programmatic_json
+          );
+
+          if (parsed.isErr()) {
+            continue; // Skip invalid data
+          }
+
+          const cdpNftData = parsed.value;
+          const interestRate = new BigNumber(cdpNftData.interest);
+          const poolDebt = new BigNumber(cdpNftData.pool_debt);
+          const collateralAddress = cdpNftData.collateral_address;
+
+          // Find matching interest rate info
+          const interestData =
+            collateralAddress ===
+            FluxConstants.collaterals.xrd.collateralAddress
+              ? xrdInterestData.entries
+              : lsulpInterestData.entries;
+
+          const interestInfo = interestData.find((entry: any) => {
+            const nodeData = entry.value.programmatic_json;
+            return (
+              nodeData &&
+              nodeData.kind === "Tuple" &&
+              new BigNumber(nodeData.fields[0].value).isEqualTo(interestRate)
+            );
+          });
```

Comment:
something like this to avoid `any`
```
const interestInfo = interestData.find((entry) => {
const nodeData = entry.value.programmatic_json;
if (nodeData.kind === "Tuple") {
    const tupleFields = nodeData.fields;
    if (tupleFields[0] && tupleFields[0].kind === "Decimal") {
        const interestRateValue = new BigNumber(tupleFields[0].value);
        return interestRateValue.isEqualTo(interestRate);
    }
}
});
```